### PR TITLE
Add cantera

### PR DESCRIPTION
**Pull request by inducer**
This recipe is derived from https://github.com/Cantera/conda-recipes. As far as changes go, I mainly dropped MKL and switched to the tarball source.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cantera/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    - {{ compiler('c') }}  # [not win]
+    - {{ compiler('cxx') }}  # [not win]
+    - vs2017_win-64 ==19.15.*  # [win]
+    - scons >=3.1
+    - python >=3.5
+    - numpy
+    - fmt
+    - eigen
+    - pywin32  # [win]
+  host:
```
Comment (isuruf) [inline recipes/cantera/meta.yaml:26]:
```suggestion
  host:
    - python >=3.5
    - numpy
    - fmt
    - eigen
    - pywin32  # [win]
```

File: recipes/cantera/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  home: https://cantera.org
+  summary: "Chemical kinetics, thermodynamics, and transport tool suite"
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: License.txt
+  dev_url: https://github.com/cantera/cantera
+  description: Cantera is an open-source suite of tools for problems involving chemical kinetics, thermodynamics, and transport processes
+  doc_url: https://cantera.org/documentation
+
+extra:
```
Comment (isuruf) [inline recipes/cantera/meta.yaml:117]: To make the git repository be https://github.com/conda-forge/cantera-feedstock
```suggestion
extra:
  feedstock-name: cantera
```

Comment (inducer) [reply recipes/cantera/meta.yaml:117]: Thx!

**Comment by isuruf** [comment]
What's the license for https://github.com/Cantera/conda-recipes ?

**Comment by inducer** [comment]
Good question. https://github.com/Cantera/conda-recipes/issues/23

**Comment by inducer** [comment]
Thanks for working on this! As for the OSX failure, I observed that *somehow* the `.dylib` seems to wind up in the Python package path:
```
2020-12-09T04:41:17.7218320Z   ERROR (cantera,lib/python3.7/site-packages/cantera/_cantera.cpython-37m-darwin.so): lib/python3.7/site-packages/cantera/build/lib/libcantera.2.5.0.dylib not found in any packages
```
Maybe scons ends up putting it there, but then nobody feels responsible for packaging it? (because it's already in `libcantera`)

**Comment by inducer** [comment]
> Remove v from version

Well-spotted, sorry!

**Comment by isuruf** [comment]
> Maybe scons ends up putting it there, but then nobody feels responsible for packaging it? (because it's already in libcantera)

scons's macos support is really bad. On OSX when libraries are built they have an install_name in the build directory and dependent libraries hard code this path. This is problematic when installing both these libraries. To fix this, the build system has to do the following,
1. After installing fix the install_name
2. If a dependent library is also installed, fix the install_name of the dependee hardcoded in the dependent.

CMake, autotools does 1&2, scons does only 1.

On top of that, in scons, the SOVERSION is 2 in Linux, but compatibility_version on OSX is 2.5

**Comment by inducer** [comment]
> On OSX when libraries are built they have an install_name in the build directory and dependent libraries hard code this path.

Oh, TIL. Thanks for explaining. (Puzzled why on Earth Apple thought this made sense, but maybe that's for another time.)

**Comment by inducer** [comment]
@bryanwweber @speth On account of your authorship of https://github.com/Cantera/conda-recipes, would you be interested in serving as co-maintainers for this recipe? See [here](https://conda-forge.org/) for more on Conda Forge, if necessary.

**Comment by bryanwweber** [comment]
@inducer Yes please, from me! Thank you for taking this on!

**Comment by inducer** [comment]
@bryanwweber Added you. Thanks for being willing to help!

**Comment by isuruf** [comment]
@bryanwweber, what's the ABI compatiblity policy of cantera C/C++ library? If I build an application with 2.4 will it run with 2.5?

**Comment by bryanwweber** [comment]
@isuruf Great question! I don't have a great answer, but I think @speth will know better than I will. The short answer is that we don't have a formal compatibility policy, and as such, I _expect_ (although I haven't checked) whether the ABI is compatible between, e.g., 2.4 and 2.5.

**Comment by inducer** [comment]
Should we postpone fixing the Windows build? I have no particular interest in it. Plus the failure seems a bit mysterious. I didn't see an error message.

**Comment by isuruf** [comment]
Yes. we can skip windows for now. Should we use BLAS/LAPACK instead of eigen?

**Comment by inducer** [comment]
> Should we use BLAS/LAPACK instead of eigen?

That may be better. The [config guide](https://cantera.org/compiling/config-options.html#scons-config) says

> Cantera can use BLAS and LAPACK libraries available on your system if you have optimized versions available (for example, Intel MKL). Otherwise, Cantera will use Eigen for linear algebra support.

which makes it sound like BLAS/LAPACK are preferred.

**Comment by inducer** [comment]
Good to go from my perspective.

**Comment by bryanwweber** [comment]
This was reported as a failure in Gentoo in https://github.com/Cantera/cantera/issues/937 as well. I haven't had time to debug, but good to know that it can be reproduced on a different system, and seems to be related to the introduction of the BLAS/LAPACK libraries. There's some more debugging info in that issue as well.

**Comment by band-a-prend** [comment]
@bryanwweber as I mentioned in initial sensitivity test issue I didn't set `blas_lapack_libs` option. I get error even for sundials being built without lapack support. I checked that building of numpy with or without lapack support doesn't affect that test result too.

@inducer 
is it ok that `system_sundials` has 2 assignments to different values in `staged-recipes/recipes/cantera/cantera_base.conf`?

**Comment by inducer** [comment]
> is it ok that `system_sundials` has 2 assignments to different values in `staged-recipes/recipes/cantera/cantera_base.conf`?

Well-spotted, thanks! I've removed the conflicting setting.

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, this is ready for review

**Review by ocefpaf — APPROVED** [review]

File: recipes/cantera/.gitattributes [hunk 3]
```diff
@@ -0,0 +1 @@
+*.bat text eol=lf
```
Comment (ocefpaf) [inline recipes/cantera/.gitattributes:1]: I don't develop on Windows so I don't really now how this affects this recipe. Does it make sense to add it to the global settings for this repo?

Comment (inducer) [reply recipes/cantera/.gitattributes:1]: Without this, git (on Linux) warned me that it had changed the line endings, which I'm not sure `cmd` would  view favorably.

Comment (jakirkham) [reply recipes/cantera/.gitattributes:1]: Yeah that’s expected. Linux would have used `lf`, which we replace with `crlf`. So git is just mentioning it did that which is fine

Comment (inducer) [reply recipes/cantera/.gitattributes:1]: OIC. In that case, I'll delete this.

Comment (isuruf) [reply recipes/cantera/.gitattributes]: We should update staged-recipes `.gitattributes` file to match that of feedstocks.

**Comment by isuruf** [comment]
Thanks for the PR and for the reviews everyone.

**Pull request merged**